File: javascript/workflow/src/verbs/noop.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { NoopArgs } from '@datashaper/schema'

import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const noopStep: ColumnTableStep<NoopArgs> = (input, { message }) => {
	console.log(message)
	input.print()
	return input
}

export const noop = stepVerbFactory(noopStep)

File: python/datashaper/datashaper/engine/verbs/noop.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
from datashaper.engine.verbs.verbs_mapping import verb

from ...table_store import TableContainer
from .verb_input import VerbInput


@verb(name="noop")
def noop(
    input: VerbInput,
    message: str,
):
    output = input.get_input()
    print(message)
    print(output.to_string())

    return TableContainer(table=output)
